### PR TITLE
Renesas: Smartbond: DMA DT does not use dma-cells

### LINKED ISSUE #73803: Renesas smartbond DMA DT does not use dma-cells
The [`renesas,smartbond-dma`](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/dma/renesas%2Csmartbond-dma.yaml) binding does not define dma-cells, and consequence of this is that renesas bindings have different ways of specifying things like dma channels associated with a node. This needs to be made coherent and align with how most of the rest of zephyr DMA DT is done, using #dma-cells and phandle array.

How I found this bug: #73747 adding SPI

Another example I found: [`renesas,smartbond-usbd`](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/usb/renesas%2Csmartbond-usbd.yaml)

### PR DESCRIPTION
**Pull request by ioannis-karachalios**
This PR should fix  #73803.

### CODE REVIEW

File: dts/bindings/dma/renesas,smartbond-dma.yaml [hunk 1]
```diff
@@ -19,3 +19,9 @@ properties:
     type: int
     const: 1
     description: Number of block counts supported
```
Comment (ydamigos) [inline dts/bindings/dma/renesas,smartbond-dma.yaml:21]: We should add `"#dma-cells"` under properties, shouldn't we?

File: dts/bindings/dma/renesas,smartbond-dma.yaml [hunk 2]
```diff
@@ -19,3 +19,9 @@ properties:
...
+
+# - #dma-cells : Must be <1>.
+# channel: the dma channel, each channel supports one DMA request line
+
+dma-cells:
+  - channel
```
Comment (ydamigos) [inline dts/bindings/dma/renesas,smartbond-dma.yaml:27]: Should we also add peripheral trigger id?

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: dts/bindings/dma/renesas,smartbond-dma.yaml (hunk 1 above)
Comment (ioannis-karachalios) [reply dts/bindings/dma/renesas,smartbond-dma.yaml:21]: Correct, we need to define it as `const` so there is no way for an overlay to change it, later on.

File: dts/bindings/dma/renesas,smartbond-dma.yaml (hunk 2 above)
Comment (ioannis-karachalios) [reply dts/bindings/dma/renesas,smartbond-dma.yaml:30]: Added

File: include/zephyr/dt-bindings/dma/dma_smartbond.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#define DMA_SMARTBOND_TRIG_MUX_UART2	0x3
+#define DMA_SMARTBOND_TRIG_MUX_I2C		0x4
+#define DMA_SMARTBOND_TRIG_MUX_I2C2		0x5
+#define DMA_SMARTBOND_TRIG_MUX_USB		0x6
+#define DMA_SMARTBOND_TRIG_MUX_UART3	0x7
+#define DMA_SMARTBOND_TRIG_MUX_PCM		0x8
+#define DMA_SMARTBOND_TRIG_MUX_SRC		0x9
+#define DMA_SMARTBOND_TRIG_MUX_GPADC	0xC
+#define DMA_SMARTBOND_TRIG_MUX_SDADC	0xD
+#define DMA_SMARTBOND_TRIG_MUX_NONE		0xF
```
Comment (ydamigos) [inline include/zephyr/dt-bindings/dma/dma_smartbond.h:29]: Let's align them.

Comment (ioannis-karachalios) [reply include/zephyr/dt-bindings/dma/dma_smartbond.h]: Fixed

**Review by ydamigos — APPROVED** [review]

**Comment by decsny** [comment]
should the USB now use these dma cells instead of the custom property?

**Comment by ioannis-karachalios** [comment]
> should the USB now use these dma cells instead of the custom property?

@decsny I'd prefer that this is handled in a separate PR as more changes are required (the driver uses the DMA accelerator directly instead of employing the corresponding API class). Please advice if a separate issue should be opened.

**Review by decsny — APPROVED** [review]

**Review by andrzej-kaczmarek — APPROVED** [review]

**Pull request merged**